### DISCUSSION TITLE
enum in Entity

### DISCUSSION THREAD

**Discussion by mrZedov** [mikro-orm/mikro-orm#4819] [Q&A]
I use NestJS. Can I create an entity with a field of type enum and so that it was a type in the PostgreSQL?
I expect that such requests will be created in the migration:

CREATE TYPE error_log_groups_code AS ENUM ('enum value 1', 'enum value 2');
'create table "error_log" ("id" serial primary key, "group" error_log_groups_code not null, "message" jsonb null);'

but in my case, it created as:

export enum ErrorLogGroupsCode {
  CommonError,
  NotificationError,
}

.....
  @Enum(() => ErrorLogGroupsCode)
  public group: ErrorLogGroupsCode;
and migrations:
'create table "error_log" ("id" serial primary key, "group" text check ("group" in ('enum value 1', 'enum value 2') not null, "message" jsonb null'  

I want to use an enum type like it works in PostgreSQL
How? Thanks )

**Comment by B4nan** [comment]
Native postgres enums are supported in v6.

**Reply by B4nan** [reply]
The pr is here: https://github.com/mikro-orm/mikro-orm/pull/4296
